src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1479,41 +1480,37 @@ public void setAllowFlight(boolean flight)
 		throw new UnimplementedOperationException();
 	}
 
+
 	@Override
 	@Deprecated
 	public void hidePlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hiddenPlayerNicks.add(player.getName());
 	}
 
 	@Override
 	public void hidePlayer(@NotNull Plugin plugin, @NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hidePlayer(player);
 	}
 
 	@Override
 	@Deprecated
 	public void showPlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hiddenPlayerNicks.remove(player.getName());
 	}
 
 	@Override
 	public void showPlayer(@NotNull Plugin plugin, @NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		showPlayer(player);
```

Comment:
This would yield an unwanted outcome.
To quote from the Spigot docs:

`If another another plugin had hidden the player too, then the player will remain hidden until the other plugin calls this method too.`